CHANGELOG.md

```diff
@@ -23,6 +23,9 @@ and this library adheres to Rust's notion of
 - `zcash_spec::PrfExpand::ARBITRARY_ZIP32_CHILD` (use `ADHOC_ZIP32_CHILD`
   instead).
 
+Note: There was no v0.2.0 release because the tag was originally pushed
+incorrectly.
```

Comment:
For future occurrences, I would prefer this be placed after the version header, not at the end, because here it looks like it's part of the "Removed" header.

Non-blocking because @daira has already created the tag for this PR before it was reviewed, and I don't want yet another tag failure.